src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrForwardFlower.cs

```diff
@@ -321,242 +321,21 @@ .. mapping.Exclude.Select(VmrPatchHandler.GetExclusionRule)
-        {
-            await TryResolvingSourceManifestConflict(repo, mappingName!, cancellationToken);
-            return true;
-        }
-
-        // Git-info file conflict
-        _logger.LogInformation("Auto-resolving conflict in {file} using PR version", filePath);
-        await repo.RunGitCommandAsync(["checkout", "--ours", filePath], cancellationToken);
-        await repo.StageAsync([filePath], cancellationToken);
-        return true;
-    }
-
-    // TODO https://github.com/dotnet/arcade-services/issues/3378: This might not work for batched subscriptions
-    private async Task TryResolvingSourceManifestConflict(ILocalGitRepo vmr, string mappingName, CancellationToken cancellationToken)
+    protected override async Task<Codeflow?> DetectRecentFlow(
+        Codeflow lastFlow,
+        Backflow? lastBackFlow,
+        ForwardFlow lastForwardFlow,
+        ILocalGitRepo repo)
     {
-        _logger.LogInformation("Auto-resolving conflict in {file}", VmrInfo.DefaultRelativeSourceManifestPath);
-
-        // We load the source manifest from the target branch and replace the
-        // current mapping (and its submodules) with our branches' information
-        var result = await vmr.RunGitCommandAsync(
-            ["show", "MERGE_HEAD:" + VmrInfo.DefaultRelativeSourceManifestPath],
-            cancellationToken);
-
-        var theirSourceManifest = SourceManifest.FromJson(result.StandardOutput);
-        var ourSourceManifest = _sourceManifest;
-        var updatedMapping = ourSourceManifest.Repositories.First(r => r.Path == mappingName);
-
-        theirSourceManifest.UpdateVersion(
-            mappingName,
-            updatedMapping.RemoteUri,
-            updatedMapping.CommitSha,
-            updatedMapping.PackageVersion,
-            updatedMapping.BarId);
-
-        var theirAffectedSubmodules = theirSourceManifest.Submodules
-            .Where(s => s.Path.StartsWith(mappingName + "/"))
-            .ToList();
-        foreach (var submodule in theirAffectedSubmodules)
+        if (lastFlow is not Backflow bf)
         {
-            theirSourceManifest.RemoveSubmodule(submodule);
+            return null;
         }
 
-        var ourAffectedSubmodules = ourSourceManifest.Submodules
-            .Where(s => s.Path.StartsWith(mappingName + "/"))
-            .ToList();
-        foreach (var submodule in ourAffectedSubmodules)
-        {
-            theirSourceManifest.UpdateSubmodule(submodule);
-        }
-
-        _fileSystem.WriteToFile(_vmrInfo.SourceManifestPath, theirSourceManifest.ToJson());
-        _sourceManifest.Refresh(_vmrInfo.SourceManifestPath);
-        await vmr.StageAsync([_vmrInfo.SourceManifestPath], cancellationToken);
+        var vmr = _localGitRepoFactory.Create(_vmrInfo.VmrPath);
+        return await vmr.IsAncestorCommit(bf.VmrSha, lastForwardFlow.VmrSha)
+            ? lastForwardFlow
```

Comment:
should this be `lastBackFlow`? Since that's the relevant flow when we're solving the conflict